src/component/TodoList.jsx

```diff
@@ -4,33 +4,51 @@ import { MdModeEdit } from "react-icons/md";
 import { AiFillDelete } from "react-icons/ai";
 import { AiOutlinePlusCircle } from "react-icons/ai";
 import TodoItems from "../Data/TodoItems";
+
 console.log(TodoItems);
 const TodoList = () => {
   const [popup, setPopup] = useState("hide");
-  const [inputValue, setInputValue] = useState("");
+  const [inputValue, setInputValue] = useState({
+    time: "",
+    date: "",
+    name: "",
+  });
   const [task, setTask] = useState(TodoItems);
+  const [editMode, setEditMode] = useState(false);
+  const [taskId, setTaskId] = useState(null);
   const submit = (e) => {
     e.preventDefault();
     setPopup("hide");
-    const newTask = {
-      id: TodoItems[TodoItems.length - 1].id + 1,
-      task: inputValue,
-    };
-    setTask([...task, newTask]);
+    if (editMode) {
+      const taskToEdit = task.find((task) => task.id === taskId);
+      taskToEdit.task = inputValue;
+      setTask([...task]);
```

Comment:
also when I edit you are setting 1 task, is edit working @anjalipaswan18

Reply:
no sir edit  is not worked